lib/fulfil/configuration.rb

```diff
@@ -9,6 +9,19 @@ class Configuration
     attr_accessor :retry_on_rate_limit
     attr_accessor :retry_on_rate_limit_wait
 
+    # Allows the client to configure a notification handler. Can be used by APM
+    # tools to monitor the number of rate limit hits.
+    #
+    # @example Use APM to monitor the API rate limit hits
+    #   Fulfil.configure do |config|
+    #     config.rate_limit_notification_handler = proc {
+    #       FakeAPM.increment_counter('fulfil.rate_limit_exceeded')
+    #     }
+    #   end
+    #
+    # @return [Proc, nil]
+    attr_accessor :rate_limit_notification_handler
+
```

Comment:
Documentation is nice. Guess we'll need to update all the existing code now 😄

Reply:
Agreed :D! Nice Friday afternoon project I guess haha.